codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/auth/http/integration/AwsSdkCustomizeSigV4Auth.java

```diff
@@ -133,6 +138,18 @@ public Map<String, Consumer<TypeScriptWriter>> getRuntimeConfigWriters(
                         );
                     }
                     return map;
+                } else {
+                    // isSigV4Service and !isAwsService are implied here.
+                    return MapUtils.of(
+                        "credentials", writer -> {
```

Comment:
I'm not up to speed, but what's the difference between configuring `credentials` here v/s `credentialDefaultProvider`? It seems more consistent to configure `credentialDefaultProvider` here. And that's what it was [before](https://github.com/aws/aws-sdk-js-v3/blob/06f51ccdc1c825f4b4de6a865520ace78c098ff3/private/weather/src/runtimeConfig.ts#L34).

Reply:
`credentialDefaultProvider` was deprecated by ID&Auth SRA. It is a field that is only generated when the AwsSdkCustomizeSigV4Auth plugin is activated by the `isAwsService` and `isSigV4Service` conditionals in conjunction, and relies on data from the (AWS) service trait.

That plugin contains the config resolver that processes the credentialDefaultProvider field along with a few other AWS customizations. Rather than bringing all of that in, I'm using a default value for the credentials field, which is functionally very similar.

Reply:
`credentialDefaultProvider` seems to be a quirk from legacy dev. Even in legacyAuth, it was simply assigned onto the `credentials` field if the credentials field was empty during config resolution.